### DISCUSSION TITLE
React hook form next js - advanced filters - OnChange - requiring 2 clicks

### DISCUSSION THREAD

**Discussion by MarvingMoreton** [react-hook-form/react-hook-form#9570] [Q&A] [answered]
Hello there,

Apologies for my question in advanced, I have been developing for less than 6 months!

I have basically built an advanced filter for my project with tags as dissimulate "checkbox" input.

However, weirdly, I have to click two times on the box for the onChange to be trigger: https://dataai-jobs.com/

I only have that issues for "checkbox" input type, here is the form:

`
<form
            onChange={handleSubmit(onChange)}
            className={classes["sidebar-form"]}
          >
          
            <div className={classes["filter-card"]}>
              <label className={classes.label} htmlFor="filter">
                Filters
              </label>
              <input
                type="search"
                placeholder="Search.."
                className={classes["input-main-filters"]}
                {...register("terms", {})}
              />
            </div>

            {/* Seniority Level */}
            <div className={classes["box-category"]}>
              <div className={classes.row}>
                <h4 className={classes.label}>Seniority</h4>
              </div>
              <div className={classes["filter-list"]}>
                {seniorities.map((optionName, i) => (
                  <React.Fragment key={optionName}>
                    <label key={optionName} className={classes.filter}>
                      <input
                        type="checkbox"
                        value={optionName}
                        name="seniority"
                        {...register("seniority", {})}
                        // className={
                        //   data.seniority.include({ optionName })
                        //     ? "activated"
                        //     : "toggledButtonId"
                        // }
                      />
                      {optionName}
                    </label>
                  </React.Fragment>
                ))}
              </div>
            </div>

            {/* remoteType */}
            <div className={classes["box-category"]}>
              <div className={classes.row}>
                <h4 className={classes.label}>Remote Work</h4>
              </div>
              <div className={classes["filter-list"]}>
                {remoteTypes.map((c, i) => (
                  <React.Fragment key={c}>
                    <label key={c} className={classes.filter}>
                      <input
                        type="checkbox"
                        value={c}
                        name="remoteTypes"
                        {...register("remoteTypes", {})}
                      />
                      {c}
                    </label>
                  </React.Fragment>
                ))}
              </div>
            </div>

            {/* workType */}
            <div className={classes["box-category"]}>
              <div className={classes.row}>
                <h4 className={classes.label}>Work Type</h4>
              </div>
              <div className={classes["filter-list"]}>
                {workTypes.map((workTypeName, i) => (
                  <React.Fragment key={workTypeName}>
                    <label key={workTypeName} className={classes.filter}>
                      <input
                        type="checkbox"
                        value={workTypeName}
                        name="workTypes"
                        {...register("workTypes", {})}
                      />
                      {workTypeName}
                    </label>
                  </React.Fragment>
                ))}
              </div>
            </div>

            {/* <CheckBoxWorkTypes control={control} /> */}

            {/* Compensation range */}
            {/* Not operational yet */}
            <SliderSalary control={control} />
            {/* Location : city or country ? */}
            {/* <CountrySelect control={control} /> */}
            {/* Tags - with "voir plus" options ? */}
            <div className={classes["filter-list"]}>
              <MuiAutoComplete control={control} />
            </div>
          </form>
`

Here's the full page's code: https://codepen.io/Marving972/pen/PoagGNP

Thanks in advanced for your help!

**Comment by MarvingMoreton** [comment, marked as answer]
Found a solution thanks to Tobias on react-hook-form's Discord :)

**Reply by getTobiasNielsen** [reply]
Using a `<form>` 's onChange caused the `target`'s event, e.g. `<some checkbox or whatever>`'s to get prevented, by `e.preventDefault()` by `handleSubmit`

Couldn't find a mention of it, in the docs, but pretty sure we had one at one point.

Also credit to @jfreedman0212 who clocked it because I were more focused on disliking the `onChange` use in the first place, for performance issues.
